etest_test/profile_test.py

```diff
@@ -0,0 +1,221 @@
+"""Tests for etest.profile."""
+
+import logging
+from typing import Any
+
+import etest.profile as sut
+
+logger = logging.getLogger(__name__)
+
+
+def test_base_arch() -> None:
+    """Ensure base arches work."""
+    for arch in ["amd64", "x86", "arm64"]:
+        result = sut.Profile(logger, False, arch, "glibc", False, True, False)
+
+        # Check profile strings
+        assert result.profile == arch
+        assert result.docker == arch
+
+        # Check arch values
+        assert result.arch == arch
+        assert result.pkg_arch == arch
+        assert result.docker_arch == arch
+
+        # Check profile options
+        assert result.hardened is False
+        assert result.multilib is True
+        assert result.systemd is False
+
+        # Check libc
+        assert result.libc == "glibc"
```

Comment:
This points out that by using a dataclass with a custom constructor we can get better testing and a better separation of data from validation.  Let me know if that makes sense or if we should dive into that a touch.

Reply:
let me know if we can talk thru this on matrix. Im not exactly sure of what implementation you are suggesting.